### PR TITLE
[WIP] Cleanup of mutation code

### PR DESCRIPTION
**Pull request by Kelenius**
#### Summary
SUMMARY: Infrastructure "Refactoring and cleanup of mutations"

#### Purpose of change
Mutation code is hella messy. Working with it is painful. I have some ideas I want to implement for it, but first I'm going to clean it up.

#### Additional context
The current plan for what I'm actually going to do is roughly:

- [x] Make use of `generic_factory`
- [ ] Standardize the use of terms "base trait", "trait", and "mutation" in function names and variables, because right now the player has functions `has_trait`, `has_base_trait`, `toggle_trait`, `set_mutation`, `unset_mutation`, `remove_mutation`, and variables `my_mutations` and `my_traits`.
- [ ] Clean up the class/function/variable names in general.
- [ ] Create a class for player mutations and put everything into that class, so the player has one variable that controls it all.
- [ ] Look into `prereqs` and `prereqs2`.
- [ ] #26105
- [ ] Other stuff.

### CODE REVIEW

**Comment by Xion350** [comment]
Really does need to be done.  It confuses me how some effects of mutations are in the mutation.json and others are all over the place.

File: src/mutation_data.cpp [hunk 1]
```diff
@@ -20,27 +21,26 @@ typedef std::set<trait_id> TraitSet;
...
+namespace
+{
+generic_factory<mutation_branch> trait_factory( "trait" );
+}
+
 std::vector<dream> dreams;
 std::map<std::string, std::vector<trait_id> > mutations_category;
 std::map<std::string, mutation_category_trait> mutation_category_traits;
-std::unordered_map<trait_id, mutation_branch> mutation_data;
+//std::unordered_map<trait_id, mutation_branch> mutation_data;
```
Comment (BevapDin) [inline src/mutation_data.cpp]: Please don not check in commented out code. Either remove it or keep it as actual code.

**Comment by Kelenius** [comment]
IRL events keep me occupied.

**Comment by ZhilkinSerg** [comment]
> 
> 
> IRL events keep me occupied.

Ping me when you'll feel you can continues with it.

**Pull request closed without merging**